src/main/java/teammates/ui/webapi/CreateAccountRequestAction.java

```diff
@@ -1,56 +1,198 @@
 package teammates.ui.webapi;
 
+import org.apache.http.HttpStatus;
+
 import teammates.common.datatransfer.attributes.AccountRequestAttributes;
 import teammates.common.exception.EntityAlreadyExistsException;
 import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Const;
 import teammates.common.util.EmailWrapper;
+import teammates.common.util.FieldValidator;
+import teammates.common.util.Logger;
+import teammates.ui.output.AccountRequestCreateErrorResults;
 import teammates.ui.output.JoinLinkData;
-import teammates.ui.request.AccountCreateRequest;
+import teammates.ui.request.AccountRequestCreateIntent;
+import teammates.ui.request.AccountRequestCreateRequest;
+import teammates.ui.request.AccountRequestType;
 import teammates.ui.request.InvalidHttpRequestBodyException;
 
 /**
  * Creates a new account request.
  */
-class CreateAccountRequestAction extends AdminOnlyAction {
+class CreateAccountRequestAction extends Action {
+
+    static final String PUBLIC_CREATE_EXISTING_ACCOUNT_REQUEST_MESSAGE =
```

Comment:
As I mentioned previously, the backend should have minimal concern about presentation details on the frontend. This is unnecessary coupling that makes it harder to work on the frontend without having to work on the backend as well. Catching the appropriate error code and displaying an appropriate error message should be enough here.

Reply:
I think currently there is not a good error return approach, so it's hard to return a customized error code to be used by frontend. I am not 100% sure but maybe there are two errors resulting in the same error code so frontend can't differentiate them.

Reply:
I will check on this later to see if existing account request is the only problem that can cause the error code. If that's the case, then it's fine to just use the error code.

Reply:
Moved the message to frontend.